syslog/syslog.c

```diff
@@ -58,13 +60,17 @@ void openlog(const char *ident, int logopt, int facility)
 {
 	if (ident != NULL) {
 		syslog_common.ident = ident;
-	} else {
+	}
+	else {
 		syslog_common.ident = argv_progname;
 	}
 
 	if (logopt & LOG_NDELAY) {
 		if (!syslog_common.open) {
-			syslog_common.logfd = open(PATH_LOG, O_WRONLY | O_NONBLOCK);
+			syslog_common.addr.sun_family = AF_UNIX;
+			strcpy(syslog_common.addr.sun_path, PATH_LOG);
+			syslog_common.addrlen = SUN_LEN(&syslog_common.addr);
+			syslog_common.logfd = socket(AF_UNIX, SOCK_DGRAM | SOCK_NONBLOCK, 0);
```

Comment:
maybe `connect` once and then use `write`? I don't know the implementation but it should be faster?

Reply:
Connect may fail if syslogd socket is not available yet (syslogd starting). What if syslogd restarts or crashes? It safer to always send to destination address.